EasyClangComplete.py

```diff
@@ -209,13 +233,15 @@ def on_query_completions(self, view, prefix, locations):
             log.debug(" showing default completions")
             return Tools.SHOW_DEFAULT_COMPLETIONS
 
-        # create a daemon thread to update the completions
-        log.debug(" starting async auto_complete at pos: %s", locations[0])
-        completion_thread = Thread(
-            target=self.completer.complete,
-            args=[view, locations[0], self.settings.errors_on_save])
-        completion_thread.deamon = True
-        completion_thread.start()
+        self.current_job_id = current_pos_id
+        log.debug(" starting async auto_complete with id: %s",
+                  self.current_job_id)
+        future = EasyClangComplete.pool_read.submit(
+                    self.completer.complete,
```

Comment:
Style nit: This is not correct indentation according to pycodestyle/pep8 linter.
Also in couple other places.

(I'm using SublimeLinter with SublimeLinter-pep8)